### PR TITLE
fix: remove trailing GitHub GET request

### PR DESCRIPTION
**Pull request by juniorbassani**
Currently, running synth version causes two GET requests to be
dispatched to GitHub in order to check for new Synth versions, wich can
be verified by running RUST_LOG=debug synth version.

This removes the additional request, and now only the child thread
spawned at the main function will be resposible for checking for
updates.

### CODE REVIEW

**Review by WiredSound — CHANGES_REQUESTED** [review]
Hey @juniorbassani , thanks for this! Looks good but I do have a minor concern - ideally I think we would want to logic to be as follows:

* User is running a Synth command that is not `synth version`
  * If they're running an old version and have not seen the message about a new version yet, display it to them (once)
  * Otherwise, execute just the command without any version message
* User runs `synth version`
  * Display the version they current have installed
  * *Always* check with GitHub (using only one HTTP request rather than two of course!) the most recent version and display the message as necessary - regardless of `"seen_versions"` in their Synth config file

In other words, I think ideally we would want `synth version` to always report about a new version.

**Comment by juniorbassani** [comment]
Thanks for the review, @WiredSound. I will follow your suggestions to fix the issues.

**Comment by juniorbassani** [comment]
@WiredSound, I believe now the PR addresses the issues you mentioned: only one GET request is made when running `synth version`, and the version command will always check for new Synth versions and display them if they exist.

Let me know you have any other concerns or suggestions.
